pkg/network/dns/stats.go

```diff
@@ -27,14 +26,21 @@ const (
 	failedResponse
 	// query means the packet contains a DNS query
 	query
-)
-
-// This const limits the maximum size of the state map. Benchmark results show that allocated space is less than 3MB
-// for 10000 entries.
-const (
+	// Subsystem name for telemetry purposes
+	dnsStatKeeperModuleName = "network_tracer_dns_stat_keeper"
+	// This const limits the maximum size of the state map. Benchmark results show that allocated space is less than 3MB
+	// for 10000 entries.
 	maxStateMapSize = 10000
 )
 
+var statsTelemetry = struct {
+	lastNumStats     telemetry.Gauge
+	lastDroppedStats telemetry.Gauge
+}{
+	telemetry.NewGauge(dnsStatKeeperModuleName, "last_num_stats", []string{}, "Gauge measuring the number of processed DNS stats"),
+	telemetry.NewGauge(dnsStatKeeperModuleName, "last_dropped_stats", []string{}, "Gauge measuring the number of dropped DNS stats"),
```

Comment:
I think better names here are `processed_stats` and `dropped_stats`. Trying to follow the guidelines here: https://prometheus.io/docs/practices/naming/